backend/src/main/java/ca/gov/dtsstn/vacman/api/data/entity/AbstractCodeEntity.java

```diff
@@ -0,0 +1,107 @@
+package ca.gov.dtsstn.vacman.api.data.entity;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import jakarta.persistence.Column;
+import jakarta.persistence.EntityListeners;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+import jakarta.validation.constraints.NotNull;
+
+/**
+ * Abstract entity for code/lookup tables with NUMERIC(6) ID constraints.
+ * Maximum ID value: 999,999
+ * Includes common effective/expiry date fields for all code tables.
+ */
+@MappedSuperclass
+@EntityListeners({ AuditingEntityListener.class })
+public abstract class AbstractCodeEntity extends AbstractAuditableEntity {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(nullable = false, updatable = false)
+	protected Long id;
+
+	@Column(name = "[EFFECTIVE_DATE]", nullable = false)
+	@NotNull(message = "Effective date is required")
+	protected LocalDateTime effectiveDate;
+
+	@Column(name = "[EXPIRY_DATE]")
+	protected LocalDateTime expiryDate;
+
+	public AbstractCodeEntity() {}
+
+	public AbstractCodeEntity(
+			@Nullable Long id,
+			@Nonnull LocalDateTime effectiveDate,
+			@Nullable LocalDateTime expiryDate) {
+		super();
+		this.id = id;
+		if (effectiveDate == null) {
+			throw new IllegalArgumentException("Effective date cannot be null");
+		}
```

Comment:
```suggestion
		Assert.notNull(effectiveDate, "effectiveDate is required; it should not be null")
```

where `Assert` is from `import org.springframework.util.Assert`